### PR TITLE
Update-SearchConsole-API-Reader

### PR DESCRIPTION
**Pull request by bdavis9725**
### Issue

The Search Console API has been updated, and thus a change to facilitate this change is required, this has been implemented in this request as per their documentation.

### Description

This change is made to make sure the codebase complies with the changes outlined in the API documentation.
Please see https://developers.google.com/search/blog/2020/12/search-console-api-updates#discovery-doc-migration for further information.

### CODE REVIEW

**Comment by bibimorlet** [comment]
The checks didn't pass. The build couldn't be done because of pip requirements issues.

File: requirements-dev.txt [hunk 1]
```diff
@@ -1,4 +1,3 @@
--r requirements.txt
```
Comment (bibimorlet) [inline requirements-dev.txt:1]: Why is this line removed?

Comment (mycaule) [reply requirements-dev.txt:1]: I removed it in the other PR it because the workflow did it in two steps.

https://github.com/artefactory/artefactory-connectors-kit/blob/c16cda81d45a4c1f9f885ad87b138716adf02c32/.github/workflows/lint_and_run_tests.yml#L42-L43

Comment (bibimorlet) [reply requirements-dev.txt:1]: Ok so there was useless redundancy am I right?

Comment (mycaule) [reply requirements-dev.txt:1]: I think the way includes in requirements.txt files are meat to be used is you define parent requirements files and derives other children files and just install the childrens.

In that case it would be just one commands:
```
# For the CI/CD (Google Actions, lint_and_run_tests.yml)
pip install -r requirements-dev.txt --upgrade --no-cache-dir --use-deprecated=legacy-resolver

# For production (Docker image, Dockerfile)
pip install -r requirements.txt --upgrade --no-cache-dir --use-deprecated=legacy-resolver
```

File: ack/readers/google_search_console/reader.py [hunk 2]
```diff
@@ -78,7 +78,7 @@ def initialize_analyticsreporting(self):
             http = credentials.authorize(httplib2.Http())
             credentials.refresh(http)
 
-            self._service = build(serviceName="webmasters", version="v3", credentials=credentials, cache_discovery=False)
+            self._service = build(serviceName="searchconsole", version="v1", credentials=credentials, cache_discovery=False)
```
Comment (mycaule) [inline ack/readers/google_search_console/reader.py:81]: LGTM @bibimorlet.
Also upgrading the dependency to `google-api-python-client==2.29.0` would be great.

**Review by bibimorlet — APPROVED** [review]
